src/components/administration/administration.routes.ts

```diff
@@ -0,0 +1,103 @@
+import { Router } from "express";
+
+import AdminController from "./administration.controller";
+
+import authentication from "../../utils/adminAuthentication";
+
+import adminAuthorization from "../../utils/adminAuthorization";
+
+import staffAuthorization from "../../utils/staffAuthorization";
+/**
+ * Router class for handling administrative routes.
+ */
+class AdminRouter {
+  public router: Router;
+
+  adminController = new AdminController();
+
+  constructor() {
+    this.router = Router();
+    this.initializeRoutes();
+  }
+  /**
+   * Initialize routes for the administration router.
+   */
+  initializeRoutes() {
+    this.router.post("/login", this.adminController.loginAdministration);
+
+    this.router.post(
+      "/addadmin",
+      authentication,
+      adminAuthorization,
+      this.adminController.addAdmin,
+    );
+
+    this.router.post(
+      "/addstudent",
```

Comment:
suggestion: you can skip the key word of add, if you want to keep that add than do separate two words with dash(-)